### PR TITLE
Update the write path to include transaction and segments id

### PR DESCRIPTION
**Pull request by frankgh**
- Only required for the HCFS profiles. There is a corresponding change on the c-side

### CODE REVIEW

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java [hunk 1]
```diff
@@ -73,16 +73,14 @@ protected Object getReader(JobConf jobConf, InputSplit split)
...
         // get compression codec
-        if (compressCodec != null) {
-            codec = HdfsUtilities.getCodec(configuration, compressCodec);
-            String extension = codec.getDefaultExtension();
-            fileName += extension;
+        CompressionCodec codec = compressCodec != null ?
+                HdfsUtilities.getCodec(configuration, compressCodec) : null;
+        if (codec != null) {
+            // append codec extension to the filename
+            fileName += codec.getDefaultExtension();
```
Comment (denalex) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java]: this code gets repeated in multiple places (even in this PR), maybe move to HdfsUtilities or some other common place ?

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/HcfsType.java [hunk 2]
```diff
@@ -45,6 +47,8 @@ public String normalizeDataSource(String dataSource) {
     // as it uses SSL for communication to Azure servers
     WASBS;
 
+    protected static final Logger LOG = LoggerFactory.getLogger(HcfsType.class.getName());
```
Comment (denalex) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/HcfsType.java]: To make is subclass specific:
```suggestion
    protected Logger LOG = LoggerFactory.getLogger(this.getClass());
```

**Comment by denalex** [comment]
There is currently a difference between `ReadBridge` and `WritableBridge` in the way how `context.setDataSource()` is handled. Will this still be required after this change ? Can we make that distinction go away, if possible ?

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java (hunk 1 above)
Comment (frankgh) [reply server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java]: Yeah, I had moved it to HcfsType.java, but then it did not feel okay for HcfsType to know anything about codecs, maybe I will move it again, or maybe we can find a better place for it

Comment (frankgh) [reply server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java]: I will move it down for now.

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/SequenceFileAccessor.java [hunk 3]
```diff
@@ -79,11 +78,12 @@ protected Object getReader(JobConf jobConf, InputSplit split) throws IOException
...
-        String filename = hcfsType.getDataUri(configuration, context);
-        LOG.debug("Filename for write without updated file extension: {}", filename);
+        String filename = hcfsType.getUriForWrite(configuration, context);
         getCompressionCodec(context);
-        filename = updateFileExtension(filename, codec);
-
+        if (codec != null) {
+            // append extension to the filename
+            filename += codec.getDefaultExtension();
+        }
```
[the remark this answers is not in the document]
Comment (frankgh) [reply server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/SequenceFileAccessor.java]: it's been added back to the common place where we build the write uri

**Comment by frankgh** [comment]
@denalex yes, I removed it in commit c327aed

**Comment by frankgh** [comment]
> What about JDBC and Ignite plugins, do they still work with corresponding C-side change ?

Integration tests are green with the corresponding C-side change. I think we should address JDBC and Ignite plugins with integration tests after merging this PR.

**Comment by frankgh** [comment]
@denalex commit ba32cc6 updates support for jdbc and ignite.

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/HcfsType.java [hunk 4]
```diff
@@ -105,6 +111,30 @@ private static void checkForConfigurationMismatch(String defaultFSScheme, String
         }
     }
 
+    /**
+     * Returns a fully resolved URI including the protocol for write
+     *
```
Comment (shivzone) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/HcfsType.java]: javadoc would complain in the absence of parameters. Also would be useful to have a short note on how the write filename is constructed in the comments

**Pull request merged**